src/agent/debuglet.ts

```diff
@@ -541,18 +626,32 @@ export class Debuglet extends EventEmitter {
               // We back-off from fetching breakpoints, and try to register
               // again after a while. Successful registration will restart the
               // breakpoint fetcher.
+              that.debuggeeRegistered = new CachedPromise();
```

Comment:
What if `this.debuggeeRegistered` is non-null here before being set to `new CachedPromise()`.  Could it be possible that the previous value of `this.debuggeeRegistered` is never resolved?

Reply:
we will resolve breakpointFetched below -- that is the promise wait on debuggeeRegistered.